backend/de.metas.business/src/main/java/de/metas/payment/api/impl/PaymentBL.java

```diff
@@ -405,13 +405,17 @@ public DefaultPaymentBuilder newBuilderOfInvoice(@NonNull final I_C_Invoice invo
 	}
 
 	@Override
-	public boolean isMatchInvoice(final I_C_Payment payment, final I_C_Invoice invoice)
+	public boolean isMatchInvoice(@NonNull final I_C_Payment payment, @NonNull final I_C_Invoice invoice)
 	{
 		final List<I_C_AllocationLine> allocations = paymentDAO.retrieveAllocationLines(payment);
 		final List<I_C_Invoice> invoices = new ArrayList<>();
 		for (final I_C_AllocationLine alloc : allocations)
 		{
-			invoices.add(alloc.getC_Invoice());
+			if(alloc.getC_Invoice_ID() > 0)
```

Comment:
(for the follow-up)

looking at the whole method, IMHO loading the invoices is not needed